hq/app/AppComponents.scala

```diff
@@ -53,14 +54,14 @@ class AppComponents(context: Context)
   //  - Regions.values() returns Chinese and US government regions that are not accessible with the same AWS account
   //  - available regions can return regions that are not in the SDK and so Regions.findName will fail
   // to solve these we return the intersection of available regions and regions.values()
-  private val availableRegions = {
-    val ec2Client = AwsClient(AmazonEC2AsyncClientBuilder.standard().withRegion(Config.region).build(), AwsAccount(stack, stack, stack, stack), Config.region)
+  private val availableRegions: List[Region] = {
+    val ec2Client = AwsClient(AmazonEC2AsyncClientBuilder.standard().withRegion(Config.region.getName).build(), AwsAccount(stack, stack, stack, stack), Config.region)
```

Comment:
I think this function can be simplified now! As the comment alludes to, there were previously two problems, in opposite directions, because there were two sources for the regions data.

*Source 1*: The SDK's hard-coded list of AWS regions
*Source 2*: Calling EC2's describe regions endpoint gives the regions that are enabled for this account

The change in the AWS SDK that caused all this trouble was to deprecate the use of a hard-coded region list in the SDK. This makes sense because although it's very conveninient, it caused exactly the kind of problems this function is here to solve. If the SDK can only call regions it knows about, then it's *essential* the SDK gets kept very very up to date. Otherwise when a new region gets added by AWS, we'd see the new region in the "enabled regions" responses, but wouldn't be able to make any SDK API calls because our out-of-date SDK doesn't know that region exists.

Now that we have switched over to the new region approach, the SDK will "have a go" with whatever region string we give it.

Previously, we were taking the *intersection* of those two source of region information, to make sure that everything always works.
- If there's a region enabled in the account that doesn't appear in the SDK, we'll skip it (newly added regions)
- If there's a region in the SDK that isn't enabled for the account, we'll skip it (China, the US Government region, and nowadays lots of regions that we exclude at the org level)

But now that the SDK doesn't need to know about regions, we can just take the enabled list of regions at face value. Whataver comes back from the question "what's enabled for this account?" can be passed straight to the SDK, and it'll have a go even if it isn't new enough to know about it.

Just what we want!

In practice this means this function can just be the result of the EC2 DescribeRegions call (with error handling), without the extra logic for intersecting with the SDK list.